### PR TITLE
Have more control over search tests

### PR DESCRIPTION
**Pull request by stsewd**
- Allow us to override the request to add HTTP_HOST.
- Don't modify the markers at run time (this doesn't play nice with .com, so now we can run the search tests in .com)

With the second change there is a design decision. Now tox -e py36 will run the search tests, and they will fail if users don't have ES installed. I can modify it so tests only run the search tests on travis (kind of hacky), or I can update the docs about how to run the tests to mention to run `tox -e py36 -- -m 'no search'` if you don't have ES installed.

### CODE REVIEW

**Comment by ericholscher** [comment]
I'd like to default to no search, and have it documented how to run tests with search.

**Comment by stsewd** [comment]
Done, by default posargs will have `-m 'not search'`. It can be overridden with the normal flow using `tox -- additional posargs` or using `TOX_POSARGS`.

File: readthedocs/search/tests/test_api.py [hunk 1]
```diff
@@ -39,7 +42,7 @@ def test_search_works_with_title_query(self, api_client, project, page_num):
             'version': version.slug,
             'q': query
         }
-        resp = api_client.get(self.url, search_params)
+        resp = self.get_search(api_client, search_params)
```
Comment (ericholscher) [inline readthedocs/search/tests/test_api.py]: I don't understand this refactor -- it seems to just make it less clear what's happening?

Comment (ericholscher) [inline readthedocs/search/tests/test_api.py]: I don't understand this refactor -- it seems to just make it less clear what's happening?

File: docs/development/tests.rst [hunk 2]
```diff
@@ -17,14 +17,26 @@ environments by running::
...
 
-    tox "'--including-search'"
+    TOX_POSARGS='' tox
+
+.. note::
+
+   If you need to override tox's posargs, but you still don't want to run the search tests,
+   you need to include ``-m 'not search'`` to your command::
+
+       tox -- -m 'not search' -x
```
Comment (humitos) [inline docs/development/tests.rst]: I liked the idea of using the `--including-search` that was before this PR, which by default added `-m 'not search'` automatically by inspecting the `-m` argument first.

If I understand correctly, now when running `tox -- -k mytest` we will be forced to add `-m 'not search'` as well.

File: readthedocs/search/tests/test_api.py (hunk 1 above)
Comment (stsewd) [reply readthedocs/search/tests/test_api.py]: This is to allow us to override the api call and add HOST https://github.com/readthedocs/readthedocs.org/pull/6630/files#diff-705f0b515b1688bcb9f2a49cabc7f1a7R12

File: docs/development/tests.rst (hunk 2 above)
Comment (humitos) [reply docs/development/tests.rst]: Can we keep this behavior?

File: docs/development/tests.rst [hunk 3]
```diff
@@ -17,14 +17,26 @@ environments by running::
...
 
 .. warning::
 
    Running tests for search needs an Elasticsearch :ref:`instance running locally <development/search:Installing and running Elasticsearch>`.
+   If you are testing inside :ref:`docker compose <development/standards:Working with Docker Compose>`,
+   you'll need to override the ``ES_HOSTS`` and ``ELASTICSEARCH_DSL`` settings to point to ``search:9200`` instead.
```
Comment (humitos) [inline docs/development/tests.rst]: Why this is not set by default when running inside docker?

File: docs/development/tests.rst (hunk 2 above)
Comment (stsewd) [reply docs/development/tests.rst]: I wasn't able to make it work with .com. And if `tox -- -k mytest` doesn't match a test from search you don't have to add `-m 'not search'`

File: docs/development/tests.rst (hunk 3 above)
Comment (stsewd) [reply docs/development/tests.rst]: We use the same settings file to run tests outside and inside docker, so if we change that we need to modify the hosts file on travis (not sure if we can do that in a simple way).

Do we have a way of knowing if we are running inside docker?

Comment (stsewd) [reply docs/development/tests.rst]: We use the same settings file to run tests outside and inside docker, so if we change that we need to modify the hosts file on travis (not sure if we can do that in a simple way).

Do we have a way of knowing if we are running inside docker?

Comment (stsewd) [reply docs/development/tests.rst]: Looks like we can set a custom host on travis https://docs.travis-ci.com/user/hosts/

Comment (stsewd) [reply docs/development/tests.rst]: Are we ok with setting the defaults to `search:9200`?

**Comment by ericholscher** [comment]
I'd like to merge this, to get the search proxy stuff working for .com -- let's create issues for any unaddressed things @humitos

**Pull request merged**